executor/adapter.go

```diff
@@ -1115,9 +1116,20 @@ func getEncodedPlan(sctx sessionctx.Context, p plannercore.Plan, genHint bool, n
 		sctx.GetSessionVars().StmtCtx.SetEncodedPlan(encodedPlan)
 	}
 	if genHint {
-		hints := plannercore.GenHintsFromPhysicalPlan(p)
-		hintStr = hint.RestoreOptimizerHints(hints)
-		sctx.GetSessionVars().StmtCtx.SetPlanHint(hintStr)
+		hints = plannercore.GenHintsFromPhysicalPlan(p)
+		sctx.GetSessionVars().StmtCtx.SetPlanHint(hints)
+	}
+
+	// remove duplicate hints
+	tableHints = make([]*ast.TableOptimizerHint, 0)
+	hintsMap := make(map[string]struct{}, len(hints))
+	for idx := range hints {
+		hintStr := hint.RestoreTableOptimizerHint(hints[idx])
+		if _, ok := hintsMap[hintStr]; ok {
+			continue
+		}
+		hintsMap[hintStr] = struct{}{}
+		tableHints = append(tableHints, hints[idx])
```

Comment:
Any tests can cover this part? Besides, is it possible to generate the duplicate hints `hints = plannercore.GenHintsFromPhysicalPlan(p)`?

Reply:
add a test `with cte as (select /*+use_index(t1, idx_ab)*/ * from t1 where a = 1) select /*+use_index(@sel_1, t1, idx_ab)*/  * from cte union select * from cte` for this